### PR TITLE
Cross polygon with iturim

### PR DESCRIPTION
**Pull request by ormauda**

### CODE REVIEW

File: src/app.test.ts [hunk 1]
```diff
@@ -1,3 +1,192 @@
-test('adds 1 + 2 to equal 3', () => {
-  expect(1+2).toBe(3);
+import { Cluster } from './types/cluster';
```
Comment (maymonyu) [inline src/app.test.ts:1]: Get all types from index.ts

File: src/app.test.ts [hunk 2]
```diff
@@ -1,3 +1,192 @@
...
+import { Coordinate } from './types/coordinate';
+import { CrossChecker } from './modules/cross-checker/cross-checker';
```
Comment (maymonyu) [inline src/app.test.ts:3]: Get CrossChecker form index,ts

File: src/app.test.ts [hunk 3]
```diff
@@ -1,3 +1,192 @@
...
+import { Itur } from './types/itur';
+import { Building } from './types';
+
+const crossChecker = new CrossChecker();
+
+function createIturExample(coordinate: Coordinate): Itur {
+    return {
+        'index': 123,
```
Comment (maymonyu) [inline src/app.test.ts:11]: Remove '' in object's fields

File: src/app.test.ts [hunk 4]
```diff
@@ -1,3 +1,192 @@
...
+test('Set3: changing location , few iturim , polygon have 4 coordinates', () => {
+    const point1 = new Coordinate(32.071341, 34.805351);
+    const point2 = new Coordinate(32.071382, 34.805109);
+    const point3 = new Coordinate(32.071223, 34.805077);
+    const point4 = new Coordinate(32.071164, 34.805324);
+    const building = new Building([point1, point2, point3, point4]);
+    const cluster = createClusterExample([building]);
+
+    const firstItur = createIturExample({ latitude: 32.071291, longitude: 34.805200 }); // inside poly
+    const secondItur = createIturExample({ latitude: 31.30161, longitude: 34.2752 }); // not inside
```
Comment (maymonyu) [inline src/app.test.ts:114]: Remove comments, we should understand it from the name of the variable

File: src/app.test.ts [hunk 5]
```diff
@@ -1,3 +1,192 @@
...
+    const thirdItur = createIturExample({ latitude: 32.071218, longitude: 34.805313 });
+    const fourthItur = createIturExample({ latitude: 32.071332, longitude: 34.805222 });
+    const crossBuildings = crossChecker.crossClusterWithIturim(cluster, [firstItur, secondItur, thirdItur, fourthItur]).geoBuildings;
+
+    expect(crossBuildings.length).toBe(1);
+    expect(crossBuildings[0].iturim ? crossBuildings[0].iturim.length : -1).toBe(4);
+    expect(crossBuildings[0].coordinates.length).toBe(4);
+});
+
+test('Set4: few iturim around the building none should be considered', () => {
```
Comment (maymonyu) [inline src/app.test.ts:143]: Change the word around to be "outside"

File: src/app.test.ts [hunk 6]
```diff
@@ -1,3 +1,192 @@
...
+    const cluster = createClusterExample([building]);
+
+    const firstItur = createIturExample({ latitude: 32.071405, longitude: 34.804975 });
+    const secondItur = createIturExample({ latitude: 32.071195, longitude: 34.804964 });
+    const thirdItur = createIturExample({ latitude: 32.071136, longitude: 34.805431 });
+    const fourthItur = createIturExample({ longitude: 32.071364, latitude: 34.805447 });
+    const crossBuildings = crossChecker.crossClusterWithIturim(cluster, [firstItur, secondItur, thirdItur, fourthItur]).geoBuildings;
+
+    expect(crossBuildings.length).toBe(1);
+    expect(crossBuildings[0].iturim ? crossBuildings[0].iturim.length : 0).toBe(0);
```
Comment (maymonyu) [inline src/app.test.ts:158]: Bug, the result is always 0...

File: src/app.test.ts (hunk 3 above)
Comment (eladyundler) [reply src/app.test.ts:11]: fixed

File: src/app.test.ts (hunk 5 above)
Comment (eladyundler) [reply src/app.test.ts:146]: fixed

File: src/types/coordinate.ts [hunk 7]
```diff
@@ -1,4 +1,9 @@
-export interface Coordinate {
-    x: number;
-    y: number;
+export class Coordinate {
```
[the remark this answers is not in the document]
Comment (ormauda) [reply src/types/coordinate.ts]: Changed to Point

**Pull request merged**